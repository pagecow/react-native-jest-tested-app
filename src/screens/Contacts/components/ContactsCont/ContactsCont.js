import React from 'react';
import { Text, View, Image, ImageBackground, ActivityIndicator } from 'react-native';
import styles from './ContactsCont.style';
import Emoji from 'react-native-emoji';
import { TouchableOpacity, TouchableWithoutFeedback } from 'react-native-gesture-handler';

export default function ContactsCont({ fav, info, navigation, setChosenContactInfo }) {
  if(!info?.length) {
    return (
      <ActivityIndicator 
        style={styles.activityIndicator}
        size="large"
        color="#6c94ad"
      />
    );
  } else {
    return (   
      <View>
        <View style={styles.subHeaderCont}>
          <Text style={styles.subHeaderText}>
            {fav === true ? 'FAVORITE CONTACTS' : 'OTHER CONTACTS'}
          </Text>
        </View>
        {info.map((e, i) => {
          return(
            <View style={styles.contactCont} key={i}>
              <TouchableWithoutFeedback 
                onPressIn={() => setChosenContactInfo(e)} 
                onPressOut={() => navigation.navigate('Details')} 
              >
                <ImageBackground 
                  style={styles.thumbnail}
                  source={{ uri: "https://static.thenounproject.com/png/2838682-200.png" }} 
                >
                  <Image 
                    style={styles.thumbnail}
                    source={{ uri: e.smallImageURL }} 
                  />
                </ImageBackground>
              </TouchableWithoutFeedback>
              <View style={styles.nameAndCompanyCont}>
                {fav === true 
                  ? <Emoji 
                      style={styles.star} 
                      name="star"
                    /> 
                  : <Emoji 
                      style={styles.starHidden} 
                      name="star"
                    /> }
                <TouchableOpacity 
                  onPressIn={() => setChosenContactInfo(e)} 
                  onPressOut={() => navigation.navigate('Details')} 
                >
                <View>
                  <Text style={styles.name}>{e.name}</Text>
                  <Text style={styles.companyName}>{e.companyName}</Text>
                </View>
                </TouchableOpacity>
              </View>
            </View>
          )
        })}
      </View>
    );
  }
};