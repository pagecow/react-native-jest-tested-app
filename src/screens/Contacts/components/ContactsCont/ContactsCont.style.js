import { StyleSheet, Dimensions } from 'react-native';

const windowHeight = Dimensions.get('window').height;

export default StyleSheet.create({
    contactCont: {
        height: windowHeight * .125,
        width: "87%",
        marginLeft: "6.5%",
        flexDirection: "row",
        alignItems: "center",
        borderBottomWidth: 1.5,
        borderBottomColor: "#f1f1f1",
    },
    companyName: {
        fontFamily: "OpenSans_400Regular",
        fontSize: 12,
        color: "#8f8f8f",
    },
    nameAndCompanyCont: {
        flexDirection: "row",
        alignItems: "flex-start",
    },
    name: {
        fontFamily: "OpenSans_600SemiBold",
        fontSize: 16,
        color: "#474747",
    },
    star: {
        fontSize: 11,
        marginTop: 3,
        marginRight: 5,
    }, 
    starHidden: {
        fontSize: 11,
        marginTop: 3,
        marginRight: 5,
        opacity: 0,
    }, 
    subHeaderCont: {
        width: "100%",
        justifyContent: "center",
        backgroundColor: "#f4f4f4",
        padding: 5,
        paddingLeft: 20,
    },
    subHeaderText: {
        fontFamily: 'OpenSans_700Bold',
        fontSize: 12,
        color: "#474747",
    },
    thumbnail: {
       height: windowHeight * .125 * .6,
       width: windowHeight * .125 * .6,
       marginRight: 30,
    }
});