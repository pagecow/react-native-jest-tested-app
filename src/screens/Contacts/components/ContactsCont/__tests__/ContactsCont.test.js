import React from 'react';
import renderer from 'react-test-renderer';

import ContactsCont from '../ContactsCont';

describe('<ContactsCont />', () => {
    it('renders without crashing', () => {
        const tree = renderer.create(<ContactsCont />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with the fav prop equal to true', () => {
        const tree = renderer.create(<ContactsCont fav="true" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('renders with the fav prop equal to false', () => {
        const tree = renderer.create(<ContactsCont fav="false" />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
