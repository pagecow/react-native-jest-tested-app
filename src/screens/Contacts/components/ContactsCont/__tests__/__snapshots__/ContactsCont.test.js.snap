// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactsCont /> renders with the fav prop equal to false 1`] = `
<ActivityIndicator
  animating={true}
  color="#6c94ad"
  hidesWhenStopped={true}
  size="large"
/>
`;

exports[`<ContactsCont /> renders with the fav prop equal to true 1`] = `
<ActivityIndicator
  animating={true}
  color="#6c94ad"
  hidesWhenStopped={true}
  size="large"
/>
`;

exports[`<ContactsCont /> renders without crashing 1`] = `
<ActivityIndicator
  animating={true}
  color="#6c94ad"
  hidesWhenStopped={true}
  size="large"
/>
`;
